docs/content/getting-started/code-generation.md

```diff
@@ -25,21 +24,21 @@ hof gen <cue, data, config>... [--flags...]
 
 
 There are two main modes for
-template based code generation.
+template-based code generation.
 
 1. `hof gen -T` is code gen from flags
```

Comment:
... is _adhoc_ code gen ...

... is _something_ code gen...

what do you think?

Reply:
couple suggestions

- Ad-hoc-based code generation can be carried out in either of two main modes.
- Two primary modes exist for generating code using ad-hoc methods.
- Ad-hoc code generation can be accomplished in one of two primary ways.
- There are two primary approaches to ad-hoc-based code generation.
- Generating code with ad-hoc methods can be done in either of two main modes.

Reply:
The location of this comment may have confused things... the change above is fine.

I'm thinking about the right-hand side of the 1. & 2. bullets, and wordsmithing the part after the is...

The two methods are ad hoc and _something_, no thinking of a good word to fill in that _something_, which is the config / re-generate-able / module based code gen (for when things get bigger or more complex)

I may be over thinking this 🤔

Reply:
In template-based code generation, two approaches dominate: ad-hoc and a more sophisticated configuration or module-based method.

what do you think?